File: app/src/test/resources/testcode-js/ClassUsagePatterns.js
// Test file to verify JavaScript class usage patterns

class BaseClass {
    static staticMethod() {
        return "static";
    }

    instanceMethod() {
        return "instance";
    }
}

class ExtendedClass extends BaseClass {
    constructor() {
        // Constructor call via super
        super();

        // Constructor call with new
        this.instance = new BaseClass();

        // Static access
        const result = BaseClass.staticMethod();
    }

    useBaseClass(param) {
        // Parameter usage
        return param.instanceMethod();
    }
}

// Variable declaration
const myBase = new BaseClass();

// Export/import pattern
export { BaseClass };

File: app/src/test/resources/testcode-cpp/class_usage_patterns.cpp
// Test file to verify C++ class usage patterns

#include <vector>
#include "class_usage_patterns.h"

class BaseClass {
public:
    static void staticMethod() {
        // static implementation
    }

    void instanceMethod() {
        // instance implementation
    }
};

class ExtendedClass : public BaseClass {
private:
    BaseClass* field;

public:
    ExtendedClass(BaseClass* param) {
        // Constructor call
        field = new BaseClass();

        // Static access
        BaseClass::staticMethod();
    }

    // Method with type annotations
    BaseClass* processBase(BaseClass* input) {
        return input;
    }

    // Template usage
    std::vector<BaseClass*> getList() {
        return std::vector<BaseClass*>();
    }
};

void usageExample() {
    // Variable declarations
    BaseClass obj;
    BaseClass* ptr = new BaseClass();
    BaseClass& ref = obj;
}
